python/ray/dag/compiled_dag_node.py

```diff
@@ -2332,31 +2345,35 @@ def visualize(
                 label += type(dag_node).__name__
                 shape = "diamond"
                 fillcolor = "red"
-
+            print("task: " + label + ", outputIdxs: " + str(task.output_idxs))
             # Add the node to the graph with attributes
             dot.node(str(idx), label, shape=shape, style=style, fillcolor=fillcolor)
-
-        # Add edges with type hints based on argument mappings
-        for idx, task in self.idx_to_task.items():
-            current_task_idx = idx
-
-            for arg_index, arg in enumerate(task.dag_node.get_args()):
-                if isinstance(arg, DAGNode):
-                    # Get the upstream task index
-                    upstream_task_idx = self.dag_node_to_idx[arg]
-
+            type_hint_type = (
+                type(dag_node.type_hint).__name__
+                if dag_node.type_hint
+                else "UnknownType"
+            ) + "\n"
+            # This logic is built on the assumption that there will only be multiple
+            # output channels if the task has multiple returns
+            # case: task with one output
+            if len(task.output_channels) == 1:
+                for downstream_node in task.dag_node._downstream_nodes:
+                    downstream_idx = self.dag_node_to_idx[downstream_node]
+                    actor_handle = task.downstream_task_idxs[downstream_idx]
                     # Get the type hint for this argument
-                    if arg_index < len(task.arg_type_hints):
-                        type_hint = type(task.arg_type_hints[arg_index]).__name__
-                    else:
-                        type_hint = "UnknownType"
-
-                    # Draw an edge from the upstream task to the
-                    # current task with the type hint
-                    dot.edge(
-                        str(upstream_task_idx), str(current_task_idx), label=type_hint
-                    )
-
+                    edge_label = type_hint_type + type(task.output_channels[0]).__name__
```

Comment:
For shard memory channels, the channel type is always `CompositeChannel`. Let's display the nested channels within so that the information will be useful. 

Ideally we can extract the exact channel from `_channel_dict`.